Sources/SwiftGRPCNIO/GRPCServerCodec.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import SwiftProtobuf
+import NIO
+import NIOFoundationCompat
+import NIOHTTP1
+
+/// Incoming gRPC package with a fixed message type.
+public enum GRPCServerRequestPart<MessageType: Message> {
+  case head(HTTPRequestHead)
+  case message(MessageType)
+  case end
+}
+
+/// Outgoing gRPC package with a fixed message type.
+public enum GRPCServerResponsePart<MessageType: Message> {
+  case headers(HTTPHeaders)
+  case message(MessageType)
+  case status(GRPCStatus)
+}
+
+/// A simple channel handler that translates raw gRPC packets into decoded protobuf messages, and vice versa.
+public final class GRPCServerCodec<RequestMessage: Message, ResponseMessage: Message> { }
+
+extension GRPCServerCodec: ChannelInboundHandler {
+  public typealias InboundIn = RawGRPCServerRequestPart
+  public typealias InboundOut = GRPCServerRequestPart<RequestMessage>
+
+  public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+    switch self.unwrapInboundIn(data) {
+    case .head(let requestHead):
+      ctx.fireChannelRead(self.wrapInboundOut(.head(requestHead)))
+
+    case .message(var message):
+      let messageAsData = message.readData(length: message.readableBytes)!
```

Comment:
Should we handle this instead of force unwrapping?

Reply:
No. When the buffer says it has this many bytes readable, then we are guaranteed to read that many.